app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningSettingsActivityTest.kt

```diff
@@ -14,110 +20,96 @@ import androidx.test.espresso.matcher.ViewMatchers.withId
 import androidx.test.ext.junit.rules.ActivityScenarioRule
 import androidx.test.ext.junit.runners.AndroidJUnit4
 import androidx.test.platform.app.InstrumentationRegistry
+import com.example.android.architecture.blueprints.todoapp.launchFragmentInHiltContainer
 import com.github.HumanLearning2021.HumanLearningApp.R
 import com.github.HumanLearning2021.HumanLearningApp.TestUtils
-import com.github.HumanLearning2021.HumanLearningApp.TestUtils.getFirstDataset
+import com.github.HumanLearning2021.HumanLearningApp.firestore.FirestoreDatabaseService
 import com.github.HumanLearning2021.HumanLearningApp.hilt.DatabaseManagementModule
 import com.github.HumanLearning2021.HumanLearningApp.hilt.Demo2Database
-import com.github.HumanLearning2021.HumanLearningApp.model.DatabaseManagement
-import com.github.HumanLearning2021.HumanLearningApp.model.DummyDatabaseManagement
-import com.github.HumanLearning2021.HumanLearningApp.model.DummyDatabaseService
+import com.github.HumanLearning2021.HumanLearningApp.model.*
 import com.schibsted.spain.barista.assertion.BaristaVisibilityAssertions.assertDisplayed
 import com.schibsted.spain.barista.interaction.BaristaClickInteractions.clickOn
 import dagger.hilt.android.testing.BindValue
 import dagger.hilt.android.testing.HiltAndroidRule
 import dagger.hilt.android.testing.HiltAndroidTest
 import dagger.hilt.android.testing.UninstallModules
 import junit.framework.AssertionFailedError
+import kotlinx.coroutines.runBlocking
 import org.hamcrest.CoreMatchers.allOf
 import org.junit.After
 import org.junit.Before
 import org.junit.Rule
 import org.junit.Test
 import org.junit.runner.RunWith
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.verify
+import java.io.File
+import java.util.*
+
+//With @UninstallModules(DatabaseManagementModule::class) -> missing binding
+//With @BindValue -> Double binding
 
 @UninstallModules(DatabaseManagementModule::class)
 @HiltAndroidTest
 @RunWith(AndroidJUnit4::class)
 class LearningSettingsActivityTest {
+
     @get:Rule
     val hiltRule = HiltAndroidRule(this)
 
-    @BindValue
-    @Demo2Database
-    val dbMgt: DatabaseManagement = DummyDatabaseManagement(DummyDatabaseService())
+    @BindValue @Demo2Database
+    val dbManagement: DatabaseManagement = DummyDatabaseManagement(DummyDatabaseService())
 
-    @get:Rule
-    val activityScenarioRule: ActivityScenarioRule<LearningSettingsActivity> = ActivityScenarioRule(
-        Intent(
-            ApplicationProvider.getApplicationContext(),
-            LearningSettingsActivity::class.java
-        ).putExtra(
-            LearningDatasetSelectionActivity.EXTRA_SELECTED_DATASET,
-            getFirstDataset(dbMgt)
-        )
-    )
+    private val datasetId = TestUtils.getFirstDataset(dbManagement).id as String
```

Comment:
```suggestion
    private val datasetId = TestUtils.getFirstDataset(dbManagement).id
```
cast looks redundant

Reply:
Yep, there were a lot of issues when I merged your string->id PR so I did it manually and this slipped through